freeglut/freeglut/src/mswin/fg_spaceball_mswin.c

```diff
@@ -42,158 +50,563 @@
+        return;
+    }
+
+    if( GetRawInputDeviceList( pRawInputDeviceList, &numDevices, 
+        sizeof(RAWINPUTDEVICELIST) ) == -1 )
+    {
+        fgError( "spaceball list" );
+        free( pRawInputDeviceList );
+        return;
+    }
+
+    for( i=0; i < numDevices; i++ )
+    {
+        RID_DEVICE_INFO devInfo;
+        unsigned int data_size = 1023;
+
+        /*
+            char device_name[1024];
+            printf("dev index: %d type: %d\n",i, pRawInputDeviceList[i].dwType);
+            if (GetRawInputDeviceInfo(
+                    pRawInputDeviceList[i].hDevice,
+                    RIDI_DEVICENAME,
+                    &device_name, &data_size) == -1)
+            {
+                fgError( "HID name" );
+                continue;
+            }
+            printf("    name: %s\n", device_name);
+        */
+
+        devInfo.cbSize = sizeof(RID_DEVICE_INFO);
+        data_size = sizeof(RID_DEVICE_INFO);
+        if( GetRawInputDeviceInfo( pRawInputDeviceList[i].hDevice,
+                RIDI_DEVICEINFO, &devInfo, &data_size ) == -1 ) 
+        {
+            fgError( "spaceball info");
+            continue;
+        }
+
+        if( devInfo.hid.dwVendorId == VENDOR_LOGI
+            || devInfo.hid.dwVendorId == VENDOR_3DX )
+        {
+            int j;
+            
+            /* found a multi-axis device with a known VID */
+            /* set defaults for potential unknown device support */
+            fg_sball_buttons = fg_sball_bitwise_bits = A3DX_DEFAULT_BUTTON_BITS;
+
+            /* set the number of valid bits and buttons based on PID */
+            for( j=0; spaceballList[j].VID > 0; j++ )
+                if( spaceballList[j].PID == devInfo.hid.dwProductId )
+                {
+                    fg_sball_bitwise_bits = spaceballList[j].bits;
+                    fg_sball_buttons = spaceballList[j].buttons;
+                }
+            /*
+                printf( "VID: 0x%04x  PID: 0x%04x  bits: %d  buttons: %d\n",
+                    devInfo.hid.dwVendorId, devInfo.hid.dwProductId, 
+                    fg_sball_bitwise_bits, fg_sball_buttons );
+            */
+        }
+    }
+    free( pRawInputDeviceList );
+}
+
+void fgSpaceballHandleWinEvent( HWND hwnd, WPARAM wParam, LPARAM lParam )
+{
+    SFG_Window* window;
     HRAWINPUT hRawInput = (HRAWINPUT)lParam;
-    UINT inputCode = (UINT)wParam;
-    UINT size;
-    BYTE *rawInputBuffer;
     PRAWINPUT pRawInput;
-    UINT res;
     RID_DEVICE_INFO sRidDeviceInfo;
+    BYTE *rawInputBuffer;
+    UINT size;
 
-    if (!fg_sball_initialized)
+    if( !fg_sball_initialized )
     {
         fgPlatformInitializeSpaceball();
-        if (!fg_sball_initialized)
-        {
-            return;
-        }
+        if( !fg_sball_initialized )
+             return;
     }
 
-    res = GetRawInputData(hRawInput, RID_INPUT, NULL, &size, sizeof(RAWINPUTHEADER));
-    if (res == -1)
+    if( GetRawInputData( hRawInput, RID_INPUT, NULL, &size,
```

Comment:
random formatting change

Reply:
hmm, this was changed to keep the style consistent with the function`fgSpaceballUpdateDeviceList()`. If you like, I can revert it, but prefer to then do the same to all 6 of the `if(... == -1) return;` statements.
verdict?

Reply:
I wanted to add that this is not just formatting, and was part of the compiled binary size (memory) optimizations.  This one saved about 100 bytes, and was part of a larger effort (that brought the binary size down to 222KB from 224KB). Note that the DLL is currently 1KB larger than before I started, which seems pretty good considering the addtional functionality.

Reply:
I don't understand. How can a formatting change affect the filesize?

Also my comments about formatting changes have nothing to do with what I prefer, or even about consistency. I just don't want to see changes in lines where you didn't change anything functional, because that muddles up the change history and makes it harder to track what each commit affected.

Reply:
checking the result of the (MSW) HID functions occurs in two functions, each requiring allocating the 'res' variable (happens in 2 functions = 2 extra allocations), copying it from the return value and then checking it (6 occurences).  I have not directly compared the compiled machine code, but in general have found that this micro-optimization does save a little (all together I guesstimate about 100 bytes or less).  Also, this was how the HID sample code I drew upon was handling the err checking.  But in the end, it makes little difference to me (and only a tiny difference to the binary size.)
your call...

Reply:
okay, i think were actually talking about two different lines of code... I believe I have applied the requested changes (plus similar).